sql/01-enable_versioning.sql

```diff
@@ -19,16 +20,26 @@ DECLARE
     v_privilege       TEXT;
 BEGIN
     SELECT
-      n.nspname, c.relname
+      n.nspname, c.relname, r.rolname
     INTO
-      p_schema, p_table
+      v_schema, v_table, v_owner
     FROM
-      pg_namespace n, pg_class c
+      pg_namespace n, pg_class c, pg_roles r
     WHERE
-      c.oid = v_table_oid
+      c.oid = p_table_oid
+    AND
+      r.oid = c.relowner
     AND
       n.oid = c.relnamespace;
 
+    -- Check that SESSION_USER is the owner of the table, or
+    -- refuse to enable versioning on this table
+    IF NOT pg_has_role(session_user, v_owner, 'usage') THEN
```

Comment:
Would we want to also be nice and allow users you have the  indirect membership role to run this function as well? Or are we forcing a SET ROLE first before running the function?

Reply:
What the code is doing is checking if the invoker user (SESSION_USER) has permissions to behave as if she was the OWNER of the table, that is it is the owner, or someone who got granted  the owner role or a superuser. So there should be no need to use `SET ROLE` as long as you do have permission to act in the target role name.

Reply:
Ok sorry. my misunderstanding of the pg_has_role function with the 'usage' option